File: C++/source/Regression/RegressionTestMessage.cpp
/***************************************************************************
 *   Copyright (C) 2005 Vidyasagara Guntaka                                *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Regression/RegressionTestMessage.h"

namespace WaveNs
{

RegressionTestMessage::RegressionTestMessage ()
    : WaveMessage (0, WAVE_OBJECT_MANAGER_TEST)
{
}

RegressionTestMessage::RegressionTestMessage (WaveServiceId serviceCode)
    : WaveMessage (serviceCode, WAVE_OBJECT_MANAGER_TEST)
{
}

RegressionTestMessage::~RegressionTestMessage ()
{
}

void RegressionTestMessage::setupAttributesForSerialization ()
{
    WaveMessage::setupAttributesForSerialization ();
}

}

File: Java/source/com/CxWave/Wave/Regression/Messages/RegressionTestMessage.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Regression.Messages;

import com.CxWave.Wave.Framework.Messaging.Local.WaveMessage;
import com.CxWave.Wave.Framework.ObjectModel.FrameworkOpCodes;
import com.CxWave.Wave.Framework.ObjectModel.Annotations.Message;
import com.CxWave.Wave.Framework.Type.WaveServiceId;
import com.CxWave.Wave.Regression.WaveTestObjectManager;

@Message (om = WaveTestObjectManager.class)
public class RegressionTestMessage extends WaveMessage
{
    public RegressionTestMessage ()
    {
        super (WaveServiceId.NullServiceId, FrameworkOpCodes.WAVE_OBJECT_MANAGER_TEST);
    }

    public RegressionTestMessage (final WaveServiceId id)
    {
        super (id);
    }
}
